packages/react/src/components/SuiteHeader/IdleLogoutConfirmationModal/IdleLogoutConfirmationModal.jsx

```diff
@@ -0,0 +1,153 @@
+  i18n: {
+    sessionTimeoutModalHeading: 'Session Timeout',
+    sessionTimeoutModalBody: 'You will be logged out due to inactivity in {countdown} seconds.',
+    sessionTimeoutModalLogoutButton: 'Log out',
+    sessionTimeoutModalStayLoggedInButton: 'Stay logged in',
+  },
+};
+
+const propTypes = {
+  /** Add class name to the rendered Modal component */
+  className: PropTypes.string,
+  /** User inactivity timeout data */
+  idleTimeoutData: PropTypes.shape(IdleLogoutConfirmationModalIdleTimeoutPropTypes),
+  /** URLs for routes used on this component */
+  routes: PropTypes.shape(IdleLogoutConfirmationModalRoutePropTypes),
+  /** Function called before any route change. Returns a Promise<Boolean>. False means the redirect will not happen. This function should never throw an error. */
+  onRouteChange: PropTypes.func,
+  /** Custom function that can be provided in case something else needs to be done when the "Stay logged in" button is clicked */
+  onStayLoggedIn: PropTypes.func,
+  /** I18N strings */
+  i18n: PropTypes.shape(IdleLogoutConfirmationModalI18NPropTypes),
+};
+
+const IdleLogoutConfirmationModal = ({
+  className,
+  idleTimeoutData,
+  routes,
+  onRouteChange,
+  onStayLoggedIn,
+  i18n,
+}) => {
+  const mergedI18N = { ...defaultProps.i18n, ...i18n };
+
+  const translate = useCallback(
+    (string, substitutions) =>
+      substitutions.reduce((acc, [key, val]) => acc.replace(key, val), string),
+    []
+  );
+
+  // This state is just to force the recreation of the IdleTimer object in the useEffect below
+  const [restartIdleTimer, setRestartIdleTimer] = useState(false);
+  // Countdown state. If countdown is greater than zero, logout confirmation dialog is going to be displayed
+  const [logoutConfirmationCountdown, setLogoutConfirmationCountdown] = useState(0);
+
+  useEffect(() => {
+    if (idleTimeoutData && routes) {
+      const timer = new IdleTimer({
+        timeout: idleTimeoutData.timeout,
+        countdown: idleTimeoutData.countdown,
+        cookieName: idleTimeoutData.cookieName,
+        cookieDomain: routes.domain,
+        onIdleTimeoutWarning: (c) => {
+          setLogoutConfirmationCountdown(c);
+        },
+        onIdleTimeout: async () => {
+          const result = await onRouteChange(
+            SuiteHeader.ROUTE_TYPES.LOGOUT,
+            routes.logoutInactivity
+          );
+          if (result) {
+            window.location.href = routes.logoutInactivity;
+          }
+        },
+        onRestart: () => {
+          setLogoutConfirmationCountdown(0);
+          onStayLoggedIn();
+        },
+      });
+
+      return () => {
+        timer.cleanUp();
+      };
+    }
+    return () => {};
+  }, [restartIdleTimer, idleTimeoutData, routes, onRouteChange, onStayLoggedIn]);
+
+  return (
+    <Modal
+      data-testid="idle-logout-confirmation"
+      className={[`${settings.iotPrefix}--session-timeout-modal`, className]
+        .filter((i) => i)
+        .join(' ')}
```

Comment:
`classnames` could be used here

Reply:
Thanks for the feedback. I've just committed the change you suggested.